cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -29,6 +29,14 @@
 #define NVCOMP_HAS_ZSTD 0
 #endif
 
+#define NVCOMP_DEFLATE_HEADER <nvcomp/deflate.h>
+#if __has_include(NVCOMP_DEFLATE_HEADER)
```

Comment:
Interesting. This is the first time I've seen `__has_include`. It's a [standardized feature since C++17](https://en.cppreference.com/w/cpp/preprocessor/include).

Reply:
Yeah, when I was adding ZSTD support, I searched libcudf for existing uses and found none :D
I'm not sure if we covered it in the C++17 talks we did a while ago.